src/renderer/components/Recent.tsx

```diff
@@ -20,6 +20,7 @@ const RecentList = () => {
   const handleClick = async (id: number) => {
     try {
       await window.connections.ipcRenderer.select(id);
+      await window.procedures.ipcRenderer.fetchProcedures();
```

Comment:
I'm not sure what this does? Fetch procedures doesn't do anything it just returns procedures and you're not assigning it to anything here?

Reply:
I understand why this works now, due to the fetchProcedures() invoking the error earlier and select not throwing an error, but I don't think we need to call fetchProcedures in here, we should change select to throw an error if the connection is not valid. I will add a commit that updates this

Reply:
like i mentioned, it's a hotfix, these should probably both be fixed in the backend though

Reply:
Oh sorry I don't know what a hotfix means

Reply:
maybe i'm wrong but i think it means a temporary fix to an important problem